x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -163,158 +147,174 @@ export const Explorer = injectI18n(
-          });
-          return;
+          this.updateExplorer(stateUpdate, true);
         }
 
-        this.updateExplorer(stateUpdate, true);
-      }
+        // Listen for changes to job selection.
+        if (action === EXPLORER_ACTION.JOB_SELECTION_CHANGE) {
+          const { selectedJobs } = payload;
+          const stateUpdate = {
+            noInfluencersConfigured: !selectedJobsHaveInfluencers(selectedJobs),
+            selectedJobs,
+          };
 
-      // RELOAD reloads full Anomaly Explorer and clears the selection.
-      if (action === EXPLORER_ACTION.RELOAD) {
-        this.props.appStateHandler(APP_STATE_ACTION.CLEAR_SELECTION);
-        this.updateExplorer({ ...payload, ...getClearedSelectedAnomaliesState() }, true);
-      }
+          this.props.appStateHandler(APP_STATE_ACTION.CLEAR_SELECTION);
+          Object.assign(stateUpdate, getClearedSelectedAnomaliesState());
 
-      // REDRAW reloads Anomaly Explorer and tries to retain the selection.
-      if (action === EXPLORER_ACTION.REDRAW) {
-        this.updateExplorer({}, false);
-      }
-    });
+          if (selectedJobs.length > 1) {
+            this.props.appStateHandler(
+              APP_STATE_ACTION.SAVE_SWIMLANE_VIEW_BY_FIELD_NAME,
+              { swimlaneViewByFieldName: VIEW_BY_JOB_LABEL },
+            );
+            stateUpdate.swimlaneViewByFieldName = VIEW_BY_JOB_LABEL;
+            // enforce a state update for swimlaneViewByFieldName
+            this.setState({ swimlaneViewByFieldName: VIEW_BY_JOB_LABEL }, () => {
+              this.updateExplorer(stateUpdate, true);
+            });
+            return;
+          }
 
-    checkboxShowChartsListener = () => {
-      const showCharts = mlCheckboxShowChartsService.state.get('showCharts');
-      const { selectedCells, selectedJobs } = this.state;
+          this.updateExplorer(stateUpdate, true);
+        }
 
-      const bounds = timefilter.getActiveBounds();
-      const timerange = getSelectionTimeRange(
-        selectedCells,
-        this.getSwimlaneBucketInterval(selectedJobs).asSeconds(),
-        bounds,
-      );
+        // RELOAD reloads full Anomaly Explorer and clears the selection.
+        if (action === EXPLORER_ACTION.RELOAD) {
+          this.props.appStateHandler(APP_STATE_ACTION.CLEAR_SELECTION);
+          this.updateExplorer({ ...payload, ...getClearedSelectedAnomaliesState() }, true);
+        }
 
-      if (showCharts && selectedCells !== null) {
-        this.updateCharts(
-          this.state.anomalyChartRecords, timerange.earliestMs, timerange.latestMs
-        );
-      } else {
-        this.updateCharts(
-          [], timerange.earliestMs, timerange.latestMs
-        );
-      }
-    };
+        // REDRAW reloads Anomaly Explorer and tries to retain the selection.
+        if (action === EXPLORER_ACTION.REDRAW) {
+          this.updateExplorer({}, false);
+        }
+      });
+
+      this.showChartsSub = mlCheckboxShowChartsService.state.watch(() => {
```

Comment:
as discussed it would be nice to eventually replace these `stateFactory` services at some point. not for this PR though.